### DISCUSSION TITLE
V should automatically add [console] to fn main in GUI Windows apps that use println

### DISCUSSION THREAD

**Discussion by medvednikov** [vlang/v#11816] [Ideas]

**Comment by dumblob** [comment]
How will that work under the hood?

I'd actually say if no `[console]` nor `[gui]` is present, then for compilation target "Windows", the **default** should be seamless support for **both** GUI and console apps (e.g. by using [ansipipe](https://github.com/robhagemans/ansipipe ) as suggested in https://github.com/vlang/v/pull/7721#issuecomment-753127523 ). If the programmer wrote explicitly `[console]`, then it'll be compiled with **no regard** of anything GUI-related. And vice versa if the programmer wrote explicitly `[gui]`, then it'll be compiled with **no regard** of anything console-related.

And actually this proposal is **solely for backwards compatibility** because since Windows 10 there is [ConPTY](https://github.com/robhagemans/ansipipe/issues/1 ).

But if the V community will not like this (which I absolutely do not understand 
as future is ConPTY which does **not** distinguish between `console/gui` so *V* would syntactically support an **already dead** technology from the first **V 1.0** release :open_mouth:), then please at least do **not** add any `[console]` to `main()` automatically without telling the user. And instead just issue a warning in case the target platform is Windows saying that any input & output will not work unless `[console]` is added.

**Comment by dumblob** [comment]
@pavi2410 thoughts?

**Comment by pavi2410** [comment]
Sorry! I am new to native development using C/C++/V. My opinion would be of no significance.

**Comment by dumblob** [comment]
Considering how Windows 10 is widespread and Win 11 is knocking on the door, I'd say we can safely **remove** the `[console]` argument and use ConPTY to be semantically on par with other big targets (unix, macOS, etc.) - i.e. do not distinguish between GUI & CLI app.
